### PR TITLE
Avoid NPE if no URL specified with DataSource type of java.sql.Driver

### PR DESCRIPTION
**Pull request by frowe**
If no URL attribute (dataSource element) or URL field (@DataSourceDefinition anno) is specified, an NPE is thrown.

### CODE REVIEW

**Comment by frowe** [comment]
#build

File: dev/com.ibm.ws.jdbc/src/com/ibm/ws/jdbc/DataSourceService.java [hunk 1]
```diff
@@ -607,7 +607,10 @@ protected void init() throws Exception {
             } else if (Driver.class.getName().equals(type)) {
                 ifc = Driver.class;
                 String url = vProps.getProperty("URL", vProps.getProperty("url"));
-                vendorImpl = jdbcDriverSvc.getDriver(url, vProps, id);
+                if (url != null && !"".equals(url)) {
```
Comment (alexsm82) [inline dev/com.ibm.ws.jdbc/src/com/ibm/ws/jdbc/DataSourceService.java]: There are two places that call getDriver - init() which you updates here and getManagedConnectionFactory().  I assume getManagedConnectionFactory should be updated with the null check as well?

**Comment by frowe** [comment]
#build

**Comment by frowe** [comment]
#build

**Comment by frowe** [comment]
None of the test failures are related to the changes for this pull request.

**Pull request merged**